src/backend/distributed/executor/partitioned_intermediate_results.c

```diff
@@ -400,24 +421,26 @@ CreatePartitionedResultDestReceiver(char *resultIdPrefix, int partitionColumnInd
  * PartitionedResultDestReceiver.
  */
 static void
-PartitionedResultDestReceiverStartup(DestReceiver *copyDest, int operation,
+PartitionedResultDestReceiverStartup(DestReceiver *dest, int operation,
 									 TupleDesc inputTupleDescriptor)
 {
-	/*
-	 * We don't expect this to be called multiple times, but if it happens,
-	 * we will just overwrite previous files.
-	 */
-	PartitionedResultDestReceiver *partitionedDest =
-		(PartitionedResultDestReceiver *) copyDest;
-	int partitionCount = partitionedDest->partitionCount;
-	for (int partitionIndex = 0; partitionIndex < partitionCount; partitionIndex++)
+	PartitionedResultDestReceiver *self = (PartitionedResultDestReceiver *) dest;
+
+	self->startupArguments.tupleDescriptor = CreateTupleDescCopy(inputTupleDescriptor);
+	self->startupArguments.operation = operation;
+
+	if (self->lazyStartup)
 	{
-		DestReceiver *partitionDest =
-			partitionedDest->partitionDestReceivers[partitionIndex];
-		if (partitionDest != NULL)
-		{
-			partitionDest->rStartup(partitionDest, operation, inputTupleDescriptor);
-		}
+		/* we are initialized, rest happens when needed*/
+		return;
+	}
+
+	/* no lazy startup, lets startup our partitionedDestReceivers */
+	for (int i = 0; i < self->partitionCount; i++)
```

Comment:
We usually keep full variable names even for loop counters  (e.g. partitionIndex)  for ease of searching and readability